akka-stream/src/main/scala/akka/stream/impl/io/FilePublisher.scala

```diff
@@ -52,6 +52,7 @@ private[akka] final class FilePublisher(f: Path, completionPromise: Promise[IORe
       chan = FileChannel.open(f, FilePublisher.Read)
     } catch {
       case ex: Exception ⇒
+        completionPromise.success(IOResult(0L, Failure(ex)))
```

Comment:
There is also one completionPromise.success in `postStop` and now there is a risk that it will fail. Perhaps change that to `trySuccess`?